### ISSUE TITLE
Add Google Lucky Search

### ISSUE THREAD

**Issue by nikitavoloboev** [deanishe/alfred-searchio#46]
I can try add this but want to wait for a new release of Searchio first to accommodate new AwGo changes.

**Comment by deanishe** [comment]
What exactly are you suggesting/requesting? What's different to #15 (which you created)?

**Comment by nikitavoloboev** [comment]
Well it was closed. I don't mind having it as separate script filters. Just wanted to keep it open so it's in the open issues.

**Comment by nikitavoloboev** [comment]
I basically want to do #16 but for new Searchio.

**Comment by deanishe** [comment]
That's pretty simple: it just requires duplicating [the regular Google definition](https://github.com/deanishe/alfred-searchio/blob/4ca2838ef9b5728f0cf8da7ac28a0c42d8ece3e0/src/lib/searchio/engines/google.json) and changing the values for `search_url`.

**Comment by nikitavoloboev** [comment]
Last time I tried building the workflow, it failed. You mentioned it wasn't a trivial change to get Searchio working with new AwGo. So I am waiting for that to complete first if that makes sense.

**Comment by deanishe** [comment]
> You mentioned it wasn't a trivial change to get Searchio working with new AwGo

I did, but that was a bit silly. It shouldn't be very hard simply because the Go component of the workflow is tiny.

**Comment by nikitavoloboev** [comment]
> It shouldn't be very hard simply because the Go component of the workflow is tiny.

Should I try do this? Or you got some spare time? 🙂

**Comment by deanishe** [comment]
I just compiled the workflow against the latest version of AwGo.

What versions are you trying to compile?

**Comment by nikitavoloboev** [comment]
```
~/clones/alfred-searchio develop
❯ modd
18:55:52: prep: go build -v -o ./src/search ./src/search.go && ./src/search google-en beast
src/search.go:28:2: cannot find package "github.com/NodePrime/jsonpath" in any of:
	/usr/local/go/src/github.com/NodePrime/jsonpath (from $GOROOT)
	/Users/nikivi/go/src/github.com/NodePrime/jsonpath (from $GOPATH)
exit status 1
^C
~/clones/alfred-searchio develop 7s
❯ go get github.com/NodePrime/jsonpath
# cd .; git clone https://github.com/NodePrime/jsonpath /Users/nikivi/go/src/github.com/NodePrime/jsonpath
Cloning into '/Users/nikivi/go/src/github.com/NodePrime/jsonpath'...
remote: Repository not found.
fatal: repository 'https://github.com/NodePrime/jsonpath/' not found
package github.com/NodePrime/jsonpath: exit status 128
```

I tried running `modd` on develop branch. NodePrime/jsonpath no longer exists and I changed references to it to the mirror. Let me try again.

**Comment by nikitavoloboev** [comment]
I read something abut vendoring dependencies which solves these kind of issues. I think it would be nice to adopt.

**Comment by deanishe** [comment]
It's not a Go package, Niki, just a single script. Vendoring doesn't work.

10 seconds with Google turns up `github.com/JumboInteractiveLimited/jsonpath`

Replace the `github.com/NodePrime/jsonpath` import with `github.com/JumboInteractiveLimited/jsonpath`.

**Comment by nikitavoloboev** [comment]
It would be cool to have CONTRIBUTING.md file on how to start developing the workflows.

I get that I have to set the three variables on running it but I fail to understand why it doesn't work for me when I did it this way.

```
~/clones/alfred-searchio develop* 8s
❯ alfred_workflow_bundleid=net.deanishe.alfred-searchio

~/clones/alfred-searchio develop*
❯ echo $alfred_workflow_bundleid
net.deanishe.alfred-searchio

~/clones/alfred-searchio develop*
❯ alfred_workflow_cache=/Users/nikivi/Library/Caches/com.runningwithcrayons.Alfred-3/Workflow\ Data/net.deanishe.alfred-searchio

~/clones/alfred-searchio develop*
❯ alfred_workflow_data=/Users/nikivi/Library/Application Support/Alfred 3/Workflow Data/net.deanishe.alfred-searchio
zsh: no such file or directory: Support/Alfred

~/clones/alfred-searchio develop*
❯ modd
19:09:52: prep: go build -v -o ./src/search ./src/search.go && ./src/search google-en beast
panic: Invalid Workflow environment: alfred_workflow_bundleid is not set, alfred_workflow_cache is not set, alfred_workflow_data is not set
```

nvm the third path being wrong, it says they were not set at all. I also tried to do it in the modd.conf script itself although that seems like the wrong way to do it. Going to google more.

**Comment by nikitavoloboev** [comment]
p.s. my own [CONTRIBUTING.md](https://github.com/nikitavoloboev/alfred-my-mind/blob/master/CONTRIBUTING.md#readme) file I reuse across all my Go workflows. I think it's the most important file for any open source project.

**Comment by deanishe** [comment]
Tbh, the only thing worth mentioning (regarding this project) is modd. Everything else is just basic Go/Python/git, which is outside the scope of the workflow.

**Comment by nikitavoloboev** [comment]
> Tbh, the only thing worth mentioning

Even then I think it's nice to have a process in place. i.e. mention that it would be nice to open an issue first before starting to work on some massive PR only for it to get rejected.

**Comment by nikitavoloboev** [comment]
> Everything else is just basic Go/Python/git

And shell knowledge I guess. Because I don't know how to set those variables so modd likes it.

**Comment by nikitavoloboev** [comment]
done #47

**Comment by deanishe** [comment]
> Because I don't know how to set those variables so modd likes it

Good point. I've added `alfred_env.sh` to the repo, which creates the expected shell environment. It's also source in `modd.conf`, so `modd` should Just Work now.

**Comment by nikitavoloboev** [comment]
I have issues getting the Google Lucky search to work.

I cloned Searchio and `git checkout develop`. I then ran your symlink script with `workflow-install -s src` and had Searchio show up in Alfred Preferences.

It says these are my installed searches

![2018-10-02 at 22 55](https://user-images.githubusercontent.com/6391776/46376419-3f0cd600-c696-11e8-80e6-ca8d4a58bea2.png)

There are no icons on them so I suppose those are not being picked up too. And these are the script filters I have:

![2018-10-02 at 22 56](https://user-images.githubusercontent.com/6391776/46376485-619eef00-c696-11e8-8927-d4a11116242a.png)

Running `Reload` seem to have done nothing as the script filters of Wiki, DDG are still there and there is no YouTube script filter at all too. Not sure what I am doing wrong.

**Comment by nikitavoloboev** [comment]
I just checked RC2 and this issues doesn't happen there. So I guess its due to me symlinking it?

**Comment by nikitavoloboev** [comment]
RC2 also has issues with icons not showing:

![2018-10-02 at 23 03](https://user-images.githubusercontent.com/6391776/46376881-7760e400-c697-11e8-96bd-d47aeddf03f0.png)

**Comment by deanishe** [comment]
What happens if you run "Reload" from the `searchio` (keyword) menu?

**Comment by nikitavoloboev** [comment]
Running reload on RC2 doesn't add the icons back. I don't mind the icons though. I just want to get google lucky search working

**Comment by deanishe** [comment]
It's probably the symlinking. Alfred doesn't notice when the workflow has changed. Try restarting Alfred.

**Comment by lukewlms** [comment]
For anyone trying to use this now: Google's "Feeling Lucky" automation is broken by a confirmation page. DuckDuckGo works though.  You can use this Alfred search syntax, with the backslash:

`https://duckduckgo.com/?q=\{query}`
